doc/contribute-docs.md

```diff
@@ -1,25 +1,78 @@
 web editor where you can propose changes to the corresponding page.
 
 Please remember to first check the [latest version](https://netplan.readthedocs.io/en/latest/)
 of our documentation and make your proposal based on that revision.
 
-## Creating a pull request
+## Contributing on GitHub
 
-If you want to follow a Git development workflow, you can also checkout the
+To follow a Git development workflow, `checkout` the
 [Netplan repository](https://github.com/canonical/netplan) and contribute your
 changes as [pull requests](https://github.com/canonical/netplan/pulls), putting
 the `documentation` label for better visibility.
 
-See the `doc/` and `examples/` directories for relevant files.
+## Directory structure
+All the documentation files are located in the `doc/` directory. The `doc/` directory contains sub-directories corresponding to different [Diátaxis](https://diataxis.fr/) sections:
+* `tutorial`
+* `explanation`
+* `howto`
+* `reference`
+
+Add new articles in the appropriate directory. You can read about [how Ubuntu implements Diátaxis for documentation](https://ubuntu.com/blog/diataxis-a-new-foundation-for-canonical-documentation).
+
+## Building the documentation
+Follow these steps to build the documentation on your local machine.
+1. Fork the [Netplan repository](https://github.com/canonical/netplan). Visit [Fork a repository](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) for instructions.
+
+2. Clone the repository to your machine:
+    ```shell
+    git clone git@github.com:your_user_name/netplan.git
+    ```
+3. Navigate to the `doc/` directory within the cloned repository:
+    ```
+    cd netplan/doc
+    ```
+
+4. Install `make` on your machine if you don't have it:
+    ```shell
+    sudo apt-get install make
+    ```
+    :::{note}
+    The `make` command is compatible with Unix systems. If you're on Windows, you can [install Ubuntu with WSL to follow along](https://github.com/canonical/open-documentation-academy/blob/main/getting-started/start_with_WSL.md).
+    :::
+
+5. Within the `doc/` directory, run the `make` command to build and serve the documentation:
+    ```shell
+    make run
+    ```
+    After you run the command, visit `http://127.0.0.1:8000` in your browser to view the local copy of the documentation.
+
+    You can find all the HTML files in the `.build/` directory.
+
+    We use the Sphinx `autobuild` module, so that any edits you make (and save) as you work are applied, and the documentation refreshes immediately.
+
+## Documentation format
+The Netplan documentation is built with Sphinx using the reStructuredText and Markdown mark-up languages. If you're new to reStructuredText, read our [reStructuredText style guide](https://canonical-documentation-with-sphinx-and-readthedocscom.readthedocs-hosted.com/style-guide/).
```

Comment:
question (non-blocking): Are we actually using reStructuredText somewhere?

Reply:
Hi, @slyon. Robert added that bit. I didn't use reStructuredText in this document, but I noticed it's used in some other documents.